src/nvim/api/buffer.c

```diff
@@ -467,7 +467,13 @@ void buffer_insert(Buffer buffer,
                    ArrayOf(String) lines,
                    Error *err)
 {
-  buffer_set_line_slice(buffer, lnum, lnum, true, false, lines, err);
+  linenr_T nlines = buffer_line_count(buffer, err);
+  if (lnum == nlines) {
+    lnum = -1;
+  }
+
+  bool end_start = lnum < 0 ? true : false;
```

Comment:
IMO, this is just as clear `bool end_start = lnum < 0;`

If you disagree that's fine; I don't want to waste time bikeshedding ;)